### PR TITLE
create static scatter plot with echarts

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase/issues/35800

### Description

Renders the basic series for the static scatter plot implemented with echarts. I also added many storybook examples for different cases that will be implemented in later PRs.

### How to verify

1. Create a basic scatter plot
2. Add to a dashboard
3. Send a subscription
4. Confirm it renders correctly

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by EmmadUsmani** [comment]
Current dependencies on/for this PR:
* master
  * **PR #34606** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34606?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35118** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35118?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34820** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34820?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #35297** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35297?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #35300** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35300?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #35301** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35301?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #35303** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35303?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #35304** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35304?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #35307** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #35321** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #35395** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35395?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #35573** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35573?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #35783** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35783?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #35996** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35996?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #35345** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #35997** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35997?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **PR #36317** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36317?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                            * **PR #35922** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35922?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #35923** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #36013** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36013?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **PR #36135** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36135?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34923** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34717** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34717?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #34772** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34772?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #34809** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34809?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #34816** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34816?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #34859** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34859?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #34869** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34869?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #34872** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34872?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #35068** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35068?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34703** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34703?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #34763** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34763?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/36317?utm_source=stack-comment).

File: dev/src/dev/render_png.clj [hunk 1]
```diff
@@ -87,9 +88,7 @@
 
 (comment
   (render-card-to-png 1)
-
-  (let [{:keys [content]} (render-pulse-card 1)]
-    (open-hiccup-as-html content))
+  (render-card-to-png 2700)
```
Comment (alxnddr) [inline dev/src/dev/render_png.clj:91]: Did this come from master?

File: frontend/src/metabase/static-viz/components/ScatterPlot/ScatterPlot.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import { calculateLegendRows } from "../Legend/utils";
+import { Legend } from "../Legend";
+import { computeStaticComboChartSettings } from "../ComboChart/settings";
+
+const WIDTH = 540;
+const HEIGHT = 360;
+const LEGEND_PADDING = 8;
+
+export function ScatterPlot({
```
Comment (alxnddr) [inline frontend/src/metabase/static-viz/components/ScatterPlot/ScatterPlot.tsx:18]: This file fully duplicates the ComboChart. With the existing implementation the static `ComboChart` can be renamed to `CartesianChart` used for the scatter one too. But if you have plans to have separate models, wdyt about extracting from here the shared part so we don't duplicate a lot?

**Review by alxnddr — APPROVED** [review]
Looks good! 
A couple of suggestion about duplication reduction which will be handy for the Waterfall static chart and about the structure

File: frontend/src/metabase/visualizations/echarts/cartesian/scatter/model.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import type { RawSeries, RowValue } from "metabase-types/api";
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/echarts/cartesian/scatter/model.ts:1]: The `scatter` folder inside `cartesian` mixes the folder structure a bit. So the idea was to have a folder per chart type with two separate folders for the model and option (and maybe events). But the `scatter` is a chart type, and it contains both model and option so it is harder to distinct what is used where by looking at the structure

**Pull request merged**